### PR TITLE
NorSand Lode angle cos

### PR DESCRIPTION
**Pull request by ezrayst**
- [x] Change definitions of lode angle to cos convention
- [x] Fix inconsistency in negative and positive within the Dmatrix

### CODE REVIEW

**Review by tianchiTJ — APPROVED** [review]

File: include/materials/norsand.tcc [hunk 1]
```diff
@@ -140,25 +140,24 @@ Eigen::Matrix<double, 6, 1> mpm::NorSand<Tdim>::compute_stress_invariants(
   Vector6d dev_stress = stress;
   for (unsigned i = 0; i < 3; ++i) dev_stress(i) -= mean_p;
 
-  // Compute J3
+  // Compute J3 (multiplied by -1 for compression positive)
```
Comment (kks32) [inline include/materials/norsand.tcc:143]: Why not just multiply by -1.0 here?

File: include/materials/norsand.tcc [hunk 2]
```diff
@@ -140,25 +140,24 @@ Eigen::Matrix<double, 6, 1> mpm::NorSand<Tdim>::compute_stress_invariants(
...
   // Compute Lode angle value
   double lode_angle_val = (3. * std::sqrt(3.) / 2.) * (j3 / std::pow(j2, 1.5));
   if (lode_angle_val > 1.0) lode_angle_val = 1.0;
   if (lode_angle_val < -1.0) lode_angle_val = -1.0;
 
   // Compute Lode angle (sin convention)
-  double lode_angle = (1. / 3.) * asin(lode_angle_val);
-  if (lode_angle > M_PI / 6.) lode_angle = M_PI / 6.;
-  if (lode_angle < -M_PI / 6.) lode_angle = -M_PI / 6.;
+  double lode_angle = (1. / 3.) * acos(lode_angle_val);
```
Comment (kks32) [inline include/materials/norsand.tcc]: Could we make it const?

**Review by kks32 — CHANGES_REQUESTED** [review]

File: include/materials/norsand.tcc (hunk 2 above)
Comment (ezrayst) [reply include/materials/norsand.tcc]: Yes

File: include/materials/norsand.tcc (hunk 1 above)
Comment (ezrayst) [reply include/materials/norsand.tcc:143]: OK

**Review by bodhinandach — APPROVED** [review]

File: include/materials/norsand.tcc [hunk 3]
```diff
@@ -406,9 +404,9 @@ void mpm::NorSand<Tdim>::compute_plastic_tensor(const Vector6d& stress,
     dR_dj3 *= std::pow(j2, -1.5);
     // Update derivative of theta in terms of R, check for sqrt of zero
     if (abs(1 - R * R) < 1.0E-6) {
-      dtheta_dR = 1.0 / 3.0 / sqrt(1.0E-6);
+      dtheta_dR = -1.0 / 3.0 / sqrt(1.0E-6);
     } else {
-      dtheta_dR = 1.0 / 3.0 / sqrt(1 - R * R);
+      dtheta_dR = -1.0 / 3.0 / sqrt(1 - R * R);
```
Comment (kks32) [inline include/materials/norsand.tcc]: Could you please use parenthesis notation, serial division is likely to be confusing

**Review by kks32 — CHANGES_REQUESTED** [review]

**Review by kks32 — APPROVED** [review]

**Pull request merged**